utils/abstract.py

```diff
@@ -0,0 +1,74 @@
+"""
+Abstract class from MHajoha's API branch

```

Comment:
Code written by someone else should probably be commited by that someone else.

Reply:
That's probably true, but then I shall ask you to decide between the DBM having an API (as a promise) or me copying over that class